dstep/driver/Application.d

```diff
@@ -186,22 +160,108 @@ private:
         return remainingArgs ~ extraArgs;
     }
 
-    bool handleDiagnostics ()
+    struct ParseFile
     {
-        bool translate = true;
+        private
+        {
+            string inputFile, outputFile, outputDirectory;
+            Index index;
+            TranslationUnit translationUnit;
+            DiagnosticVisitor diagnostics;
+
+            Language language;
+            string[] argsToRestore;
+            Compiler compiler;
+            string[] compilerArgs;
+            Arguments arguments;
+        }
+
+        this(Language language, string[] argsToRestore, string fileName, bool createOutputFileName, Arguments arguments,
+             Compiler compiler, string[] compilerArgs, string outputDir)
+        {
+            this.language = language;
+            this.argsToRestore = argsToRestore.dup;
+            this.arguments = arguments;
+            this.compiler = compiler;
+            this.compilerArgs = compilerArgs.dup;
+            
+            inputFile = fileName;
+
+            if(outputDir != null)
+                outputFile = buildPath(outputDir, defaultOutputFilename());
+            else if(!createOutputFileName && arguments.output != "")
```

Comment:
There are still several cases when there's a missing space between an `if` and the opening parenthesis.